File: src/main/kotlin/leetcode/MajorityElement.kt
package leetcode

class MajorityElement {
    fun majorityElement(nums: IntArray): Int {
        var majorityElement = nums[0]
        var count = 1
        for (num in nums) {
            if (majorityElement == num) {
                count++
            } else {
                count--
                if (count == 0) {
                    majorityElement = num
                }
            }
        }
        return majorityElement
    }
}

File: leetcode/src/main/cpp/MajorityElement.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 1;
        int mj = nums[0];
        for (auto& num : nums) {
            if (count == 0) {
                mj = num;
                count = 1;
            } else if (mj == num) {
                count++;
            } else {
                count--;
            }            
        }
        return mj;
    }
};